packages/@aws-cdk/cdk/lib/cloudformation/stack.ts

```diff
@@ -533,6 +547,19 @@ export class Stack extends Construct {
       }
     }
   }
+
+  /**
+   * Calculcate the stack name based on the construct path
+   */
+  private calculateStackName() {
+    // In tests, it's possible for this stack to be the root object, in which case
+    // we need to not ignore it.
+    const rootPath = this.node.scope !== undefined ? this.node.rootPath() : [this];
+
+    return rootPath
+        .map(c => c.node.id)
```

Comment:
I would rather we use `uniqueId` as the stack name by default. `id` can accept almost any character and stack names are limited. `uniqueId` is safe.

Reply:
UniqueId will add the hash though, and I didn't want that. We're going to have to have *some* stable and predictable IDs for people to interact with, if they're going to have to accept that we will generate resource IDs at random. We're already getting complaints about `Output`s being unpredictable and hard to consume from scripts, let's not make it worse by doing the same to stack names.

Reply:
I am okay with special cases root stacks (same as for logical IDs) but I honestly don't think think there is another way. Think of a situation where a stack is buried somewhere within the 10 layers of constructs with random IDs with all kinds of characters and unlimited length. The stack ID will *not* be valid.

Reply:
We still have a middle ground, we can trim down the ID without appending the hash.

Reply:
That’s not going to be enough to ensure uniqueness in all cases.

I feel this is very similar to any resource physical name that’s generated by CFN and users have the controls to specify a stack name manually if they want

Reply:
Okay. I will disagree and commit and send all complaints about it your way :)

Reply:
Should we make `uniqueId` implement this heuristic about root IDs not being mangled?